hack/upgrade-test-operator-sdk.sh

```diff
@@ -0,0 +1,315 @@
+#!/bin/bash -e
+#
+# This file is part of the KubeVirt project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2023 Red Hat, Inc.
+#
+# Usage:
+# make upgrade-test-operator-sdk
+#
+# Use Openshift-CI "optional-operators-ci-*" workflow to:
+# - Use the operator-sdk to upgrade a pre deployed bundle; use a new bundle,
+#   named 100.0.0 with the contents of the open PR (this can include new
+#   dependent images, new CRDs...).
+# - the script then performs various validations against the upgraded version.
+
+MAX_STEPS=$(( $(grep -c "Msg " "$0") - 2)) # subtract self line and the function name
+CUR_STEP=1
+HCO_DEPLOYMENT_NAME=hco-operator
+HCO_WH_DEPLOYMENT_NAME=hco-webhook
+HCO_NAMESPACE="kubevirt-hyperconverged"
+HCO_KIND="hyperconvergeds"
+HCO_RESOURCE_NAME="kubevirt-hyperconverged"
+TARGET_VERSION=100.0.0
+VMS_NAMESPACE=vmsns
+
+OUTPUT_DIR=${OUTPUT_DIR:-_out}
+
+echo "INITIAL_VERSION: $INITIAL_VERSION"
+
+function Msg {
+    { set +x; } 2>/dev/null
+    echo "--"
+    for a in "$@"; do
+        echo "Upgrade Step ${CUR_STEP}/${MAX_STEPS}: $a"
+    done
+    echo "--"
+    ((CUR_STEP += 1))
+    set -x
+}
+
+export CMD="oc"
+
+echo "oc version"
+${CMD} version || true
+
+function cleanup() {
+    rv=$?
+    if [ "x$rv" != "x0" ]; then
+        echo "Error during upgrade: exit status: $rv"
+        make dump-state
+        echo "*** Upgrade test failed ***"
+    fi
+    exit $rv
+}
+
+trap "cleanup" INT TERM EXIT
+
+source hack/compare_scc.sh
+OUTPUT_DIR=${OUTPUT_DIR} dump_sccs_before
+
+CSV=$( ${CMD} get csv -o name -n ${HCO_NAMESPACE} | grep "kubevirt-hyperconverged-operator")
+
+Msg "Check that cluster is operational before upgrade"
+timeout 10m bash -c 'export CMD="${CMD}";exec ./hack/check-state.sh'
+
+${CMD} get subscription -n ${HCO_NAMESPACE} -o yaml
+${CMD} get pods -n ${HCO_NAMESPACE}
+
+Msg "Images before upgrade"
+${CMD} get deployments -n ${HCO_NAMESPACE} -o yaml | grep image | grep -v imagePullPolicy
```

Comment:
nit: this can be done more elegantly/cleanly with:
`${CMD} get deployments -n ${HCO_NAMESPACE} -o json | jq -r .items[].spec.template.spec.containers[].image`